source/common/grpc/google_async_client_impl.cc

```diff
@@ -182,7 +182,16 @@ void GoogleAsyncStreamImpl::initialize(bool /*buffer_body_for_retry*/) {
   ctxt_.set_deadline(abs_deadline);
   // Fill service-wide initial metadata.
   auto initial_metadata = Http::RequestHeaderMapImpl::create();
-  parent_.metadata_parser_->evaluateHeaders(*initial_metadata, options_.parent_context.stream_info);
+  // TODO(cpakulski): Find a better way to access requestHeaders after runtime guard
+  // envoy_reloadable_features_unified_header_formatter runtime guard is deprecated.
+  // Maybe put it to parent_context?
+  parent_.metadata_parser_->evaluateHeaders(
+      *initial_metadata,
+      ((options_.parent_context.stream_info != nullptr) &&
+       (options_.parent_context.stream_info->getRequestHeaders() != nullptr))
+          ? *options_.parent_context.stream_info->getRequestHeaders()
+          : *Http::StaticEmptyHeaders::get().request_headers,
+      options_.parent_context.stream_info);
```

Comment:
And here.

Reply:
used `evaluateHeaders` wrapper with 2 params.